examples/record-playback/rs-record-playback.cpp

```diff
@@ -0,0 +1,268 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#include <librealsense2/rs.hpp> // Include RealSense Cross Platform API
+#include "example.hpp"          // Include short list of convenience functions for rendering
+#include <chrono>
+
+#include <imgui.h>
+#include "imgui_impl_glfw.h"
+
+// Includes for time display
+#include <sstream>
+#include <iostream>
+#include <iomanip>
+
+
+// Helper function for dispaying time conveniently
+std::string pretty_time(std::chrono::nanoseconds duration);
+// Helper function for rendering a seek bar
+void draw_seek_bar(rs2::playback& playback, int* seek_pos, float2& location, float width);
+
+int main(int argc, char * argv[]) try
+{
+    // Create a simple OpenGL window for rendering:
+    window app(1280, 720, "RealSense Record and Playback Example");
+    ImGui_ImplGlfw_Init(app, false);
+
+	// Create booleans to control GUI (recorded - allow play button, recording - show 'recording to file' test)
+	bool recorded = false;
+	bool recording = false;
+
+	// Declare a texture for the depth image on the GPU
+    texture depth_image;
+
+	// Declare frameset and frames which will hold the data from the camera
+	rs2::frameset frames;
+	rs2::frame depth;
+
+    // Declare depth colorizer for pretty visualization of depth data
+    rs2::colorizer color_map;
+
+	// Create a shared pointer to a pipeline
+    auto pipe = std::make_shared<rs2::pipeline>();
+	
+	// Start streaming with default configuration
+    pipe->start();
+	
+	// Initialize a shared pointer to a device with the current device on the pipeline
+	auto device = std::make_shared<rs2::device>(pipe->get_active_profile().get_device());
```

Comment:
rs2::device already acts as a smart pointer, and the recorder/playback derive from it. No need to use `shared_ptr`...

Reply:
I'm not 100% sure, but can you assign null to device / pipeline to reset them?
Speaking of `shared_ptr`, take a look at [context.h](https://github.com/IntelRealSense/librealsense/pull/1533/commits/f395f74f0c0608da5e3f61facac6e9ae3b387ca9#diff-25574220b8a6cbf9568f614ff18deb15R147), we found a fascinating bug someone left behind ;)

Reply:
I am referring to device, not sure you can do that with pipeline.
And yeah I saw that...some Easter eggs for ya :), when does it happen though? If I remember right the map is not a cache, rather the actual storage...I might

Reply:
Open playback device, destroy playback device, open record to same file, but the file is still open for reading. 
This was due to cycle of `shared_ptr`s starting from context and going into playback device. 

Regarding the demo, since it mixes heavily reading and writing into the same file, it is important we communicate how to let go of existing recorder or playback (including via pipeline). `dev = nullptr; pipe = nullptr;` can be also readable, but I wonder if it wont be confusing to users (these are stack objects after all)